### PR TITLE
Warnings fixed, sfreerdp allow to specify certificate

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: libfreerdp/core/transport.h [hunk 1]
```diff
@@ -113,8 +113,8 @@ FREERDP_LOCAL BOOL transport_set_connected_event(rdpTransport* transport);
 FREERDP_LOCAL BOOL transport_set_recv_callbacks(rdpTransport* transport, TransportRecv recv,
                                                 void* extra);
 
-FREERDP_LOCAL transport_tcp_connect(rdpTransport* transport, const char* hostname, int port,
-                                    DWORD timeout);
+FREERDP_LOCAL int transport_tcp_connect(rdpTransport* transport, const char* hostname, int port,
+                                        DWORD timeout);
```
Comment (hardening) [inline libfreerdp/core/transport.h:117]: good catch

File: winpr/include/winpr/synch.h [hunk 2]
```diff
@@ -293,9 +293,18 @@ extern "C"
 	InitializeCriticalSectionAndSpinCount(lpCriticalSection, dwSpinCount)
 #endif
 
+#if defined(__clang__)
```
Comment (hardening) [inline winpr/include/winpr/synch.h:296]: what about putting this enter block and exit block in a specific header and just include it ?

**Review by mfleisz — APPROVED** [review]
LGTM - agree that putting the `#pragma`s in a separate header file would be a nice idea

File: winpr/include/winpr/synch.h (hunk 2 above)
Comment (akallabeth) [reply winpr/include/winpr/synch.h:296]: good idea, so we could catch compiler differences better.

**Comment by akallabeth** [comment]
I´ll merge this and add the silence warning macro in another one.

**Pull request merged**